contrib/utilities/indent-branch

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash
+## ---------------------------------------------------------------------
+##
+## Copyright (C) 2018 by the deal.II authors
+##
+## This file is part of the deal.II library.
+##
+## The deal.II library is free software; you can use it, redistribute
+## it, and/or modify it under the terms of the GNU Lesser General
+## Public License as published by the Free Software Foundation; either
+## version 2.1 of the License, or (at your option) any later version.
+## The full text of the license can be found in the file LICENSE at
+## the top level of the deal.II distribution.
+##
+## ---------------------------------------------------------------------
+
+#
+# This script does the same thing as contrib/utilities/indent but only reformats
+# files which have changed (or been added and not committed) since the last
+# commit to the master branch. It should be run as
+#
+#    make indent-branch
+#
+# from a build directory.
+#
+
+if test ! -d source -o ! -d include -o ! -d examples ; then
+  echo "*** This script must be run from the top-level directory of deal.II."
+  exit 1
+fi
+
+# Add the location 'download_clang_format' or 'compile_clang_format'
+# installs clang-format to to the local PATH.
+CLANG_FORMAT_PATH="$(cd "$(dirname "$0")" && pwd)/programs/clang-6/bin"
+export PATH="${CLANG_FORMAT_PATH}:${PATH}"
+
+if ! [ -x "$(command -v clang-format)" ]; then
+  echo "*** No clang-format program found."
+  echo "***"
+  echo "*** You can run the 'download_clang_format' script or the"
+  echo "*** 'compile_clang_format' script for obtaining a compatible"
+  echo "*** binary and installing to an appropriate directory."
+  exit 1
+fi
+
+# Make sure to have the right version. We know that clang-6.0.0
+# and clang-6.0.1 work. Hence, test for clang-6.0.
+CLANG_FORMAT_VERSION="$(clang-format --version)"
+CLANG_FORMAT_MAJOR_VERSION=$(echo "${CLANG_FORMAT_VERSION}" | sed 's/^[^0-9]*\([0-9]*\).*$/\1/g')
+CLANG_FORMAT_MINOR_VERSION=$(echo "${CLANG_FORMAT_VERSION}" | sed 's/^[^0-9]*[0-9]*\.\([0-9]*\).*$/\1/g')
+
+if [ "${CLANG_FORMAT_MAJOR_VERSION}" -ne 6 ] || [ "${CLANG_FORMAT_MINOR_VERSION}" -ne 0 ]; then
+  echo "*** Found a version of clang-format different than the required version 6.0."
+  exit 1
+fi
+
+# we cannot store nul-delimited output in shell variables so store file names as
+# a base64 string.
+
+# add any new files:
+UNSORTED_FILES=$(git ls-files -z --others --exclude-standard | base64 | tr -d '\n')
+# add any files changed relative to the last commit to master:
+UNSORTED_FILES+=$(git diff -z --name-only master | base64 | tr -d '\n')
+SORTED_FILES=$(base64 -d <<< $UNSORTED_FILES | sort -z | uniq -z | base64)
```

Comment:
I don't think I understand all of this. Why do you do all of this `base64` and `base64 -d`? And why is the `tr -d` necessary?

If your goal is to translate `\0` into a newline, just do ` | perl -p -e 's/\0/\n/g;'`

Reply:
The `tr -d` is not necessary; I'll remove it.

I will put an expanded version of this comment in the script itself, but the problem we have is that since `\n` is a valid filename character we cannot also use it as a delimiter. Hence (the other indentation script does this too) we have to store everything as `NUL` delimited output (hence all the `-z` flags). However, bash cannot store a string with multiple `NUL` characters in it, so I convert it to base64 as an intermediate step purely to make bash happy.

Reply:
So you do all this because you want to support file names that contain `\n`? I mean, it's unlikely both that we ever want to use this, or that anyone else who uses the script ever wants to do this. 

But I'm also ok with keeping things if that's what you want to d.